discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -50,6 +52,78 @@ def list_nodes_with_name_filter(self, name_filter) -> List[libvirt.virDomain]:
     def list_networks(self):
         return self.libvirt_connection.listAllNetworks()
 
+    @staticmethod
+    def _list_disks(node):
+        all_disks = minidom.parseString(node.XMLDesc()).getElementsByTagName('disk')
+
+        results = []
+
+        for disk_xml in all_disks:
+            results.append(LibvirtController.to_disk_obj(disk_xml))
+
+        return results
+
+    @staticmethod
+    def to_disk_obj(disk_xml):
```

Comment:
`s/to_disk_obj/disk_xml_to_disk_obj/`

since this is a method of the libvirt *controller*, the name `to_disk_obj` might be a bit misleading